kotsadm/operator/pkg/client/deploy.go

```diff
@@ -54,17 +54,29 @@ func (c *Client) diffAndRemovePreviousManifests(applicationManifests Application
 	}
 
 	// we need to find the gvk+names that are present in the previous, but not in the current and then remove them
+	// namespaces that were remoed from YAML and added to additionalNamespaces should not be removed
 	decodedPreviousStrings := strings.Split(string(decodedPrevious), "\n---\n")
 	decodedPreviousMap := map[string]string{}
+
+DecodedPrevious:
 	for _, decodedPreviousString := range decodedPreviousStrings {
-		decodedPreviousMap[GetGVKWithNameAndNs([]byte(decodedPreviousString), targetNamespace)] = decodedPreviousString
+		k, o := GetGVKWithNameAndNs([]byte(decodedPreviousString), targetNamespace)
+		if o.APIVersion == "v1" && o.Kind == "Namespace" {
+			for _, n := range applicationManifests.AdditionalNamespaces {
+				if o.Metadata.Name == n {
+					continue DecodedPrevious
```

Comment:
i don't see why `DecodedPrevious` is needed. isn't simply `continue` enough? also, `decodedPreviousStrings` isn't being modified, so how does going to `DecodedPrevious` work here? won't that cause an infinite loop?